### PR TITLE
Update types.js so that it is ES5 compatible

### PR DESCRIPTION
**Pull request by xaviervia**
Hello! Thanks for a great library.

We are trying to use this library in IE (11), and we would like not to have to polyfill to be able to use it. This pull request reimplements `some` and `find` for the `CardTypes` so that they don’t rely on the ES6 browser implementation.

Let me know if this is something that works for you.

### CODE REVIEW

**Comment by bendrucker** [comment]
Hi! Couple things:

*  `Array.prototype.some` is part of ES5
* I'm typically inclined to embrace polyfilling
  * Happy to learn why you want to avoid it
  * Not polyfilling means every user pays a small cost (extra code) for backwards compatibility
  * You can choose to polyfill, they don't get a choice
* Even if we do make this strict ES5, use a [package](https://github.com/stefanduberg/array-find)
  * Tests/docs guarantee it works to spec
  * module isolation makes it easy to remove in the future

**Comment by xaviervia** [comment]
Hi @bendrucker , thanks for answering!

* You are right, of course, I reverted the change for `some` since it is unnecessary

I do agree in general terms with polyfilling, it is not great for libraries to have to stuck with old APIs; however we have the problem that we are using this library as a dependency of _another library_, and our current consumers are not required to polyfill. We found your library to be the best one for creditcard detection, and we thought that being that there is just a single function call that makes it not ES5 compatible, it would make sense to make the change there. We are not in a position to tell all our consumers to polyfill, so if we can’t we will have to maintain a fork :D (or go with other library, but we rather don’t do that since this one is excellent).

In this particular case my feeling is that the overhead of using `reduce` instead of `find` is small enough that justifies not having to polyfill / maintain a fork.

**Comment by xaviervia** [comment]
Right. I think I completely missed the second comment. Would you prefer to have the `array-find` package instead of the custom implementation? I can replace it :)

**Comment by bendrucker** [comment]
Got it, that makes sense. Transitively asking users to polyfill is more effort and error prone so I think maintaining strict ES5 compatibility makes sense. I do want to make sure CI verifies that. Adding node 0.12 to travis should take care of that.

I'd definitely prefer a package over including something inline for the reasons described earlier.

**Comment by xaviervia** [comment]
Perfect. I updated it to use `array-find` and incorporate 0.12 in Travis.

Unfortunately it seems the `tap-difflet` is using ES6 features so [tests in node 0.12 can’t run](https://travis-ci.org/bendrucker/creditcards/jobs/362975886#L530). 

I’ll think how we could test this and come back with more updates.

File: types.js [hunk 1]
```diff
@@ -18,20 +19,7 @@ function CardTypes (types) {
   }
 
   function find (test) {
```
Comment (bendrucker) [inline types.js]: No need to define `find`. 

```
var find = require('array-find')
# later
return {find: find.bind(null, types)}
```

File: package.json [hunk 2]
```diff
@@ -28,6 +28,7 @@
     "tape": "^4.0.0"
   },
   "dependencies": {
+    "array-find": "^1.0.0",
```
Comment (bendrucker) [inline package.json]: ~ for consistency even though this won't be updated

**Comment by bendrucker** [comment]
Hey, just checking on the last two small changes I requested. Feel free to remove tap-difflet (or try to find an older version). It's included to prettify the test output and is certainly not required.

**Comment by xaviervia** [comment]
@bendrucker all updated! Sorry for the delay, we had some sick time last week :(

**Pull request merged**